src/main/java/com/google/api/codegen/transformer/php/PhpPackageMetadataNamer.java

```diff
@@ -21,12 +21,40 @@
 /** PHPPackageMetadataNamer provides PHP specific names for metadata views. */
 public class PhpPackageMetadataNamer extends PackageMetadataNamer {
   private Name serviceName;
-  private String domainLayerLocation;
+  private String metadataIdenfifier;
 
   public PhpPackageMetadataNamer(String packageName, String domainLayerLocation) {
-    // Get the service name from the package name by removing the version suffix (if any).
-    this.serviceName = getApiNameFromPackageName(packageName);
-    this.domainLayerLocation = domainLayerLocation;
+    String packageNameWithoutVersion = PhpPackageUtil.getPackageNameBeforeVersion(packageName);
+    String[] packagePieces = PhpPackageUtil.splitPackageName(packageNameWithoutVersion);
+
+    this.serviceName = Name.upperCamel(packagePieces);
+
+    // To build metadata identifier, update pieces to be lowercase, so each piece is treated as a
+    // single piece in the Name object
+    for (int i = 0; i < packagePieces.length; i++) {
+      packagePieces[i] = packagePieces[i].toLowerCase();
+    }
+
+    Name composerVendor;
+    Name composerPackage;
+    // The metadataIdentifier for composer is formatted as "vendor/package". If a
+    // domainLayerLocation is provided, set that as the vendor. Otherwise, take the first piece of
+    // packagePieces to use as the vendor.
+    if (domainLayerLocation != null && !domainLayerLocation.equals("")) {
+      composerVendor = Name.anyCamel(domainLayerLocation);
+      composerPackage = Name.anyLower(packagePieces);
+    } else if (packagePieces.length == 1) {
+      composerVendor = Name.anyLower(packagePieces[0]);
+      composerPackage = composerVendor;
+    } else {
+      composerVendor = Name.anyLower(packagePieces[0]);
+      composerPackage = Name.from();
+      for (int i = 1; i < packagePieces.length; i++) {
+        composerPackage = composerPackage.join(Name.anyLower(packagePieces[i]));
+      }
+    }
+    metadataIdenfifier =
+        composerVendor.toSeparatedString("-") + "/" + composerPackage.toSeparatedString("-");
```

Comment:
Just in case: is it ok to use slash here explicitly (`"/"`)? I'm asking because there are abstractions around naming in gapic-generator, which abstract the actual separators used depending on a language (slash looks like a separator).

Reply:
It is a separator in the package name. Maybe there should be a function in surface namer that contains this detail? I'm not sure. This is a php-specific class, so maybe it is acceptable to have php-specific formatting logic here.